test/unit/xfa_tohtml_spec.js

```diff
@@ -588,4 +588,45 @@ describe("XFAFactory", function () {
     expect(a.value).toEqual("qwerty/");
     expect(a.attributes.href).toEqual("");
   });
+
+  it("should replace button with an URL by a link", function () {
+    const xml = `
+<?xml version="1.0"?>
+<xdp:xdp xmlns:xdp="http://ns.adobe.com/xdp/">
+  <template xmlns="http://www.xfa.org/schema/xfa-template/3.3">
+    <subform name="root" mergeMode="matchTemplate">
+      <pageSet>
+        <pageArea>
+          <contentArea x="123pt" w="456pt" h="789pt"/>
+          <medium stock="default" short="456pt" long="789pt"/>
+        </pageArea>
+      </pageSet>
+      <subform name="first">
+        <field y="1pt" w="11pt" h="22pt" x="2pt">
+          <ui>
+            <button/>
+          </ui>
+          <event activity="click" name="event__click">
+            <script contentType="application/x-javascript">
+              app.launchURL("https://github.com/mozilla/pdf.js", true);
+            </script>
+          </event>
+        </field>
+      </subform>
+    </subform>
+  </template>
+  <xfa:datasets xmlns:xfa="http://www.xfa.org/schema/xfa-data/1.0/">
+    <xfa:data>
+    </xfa:data>
+  </xfa:datasets>
+</xdp:xdp>
+    `;
+    const factory = new XFAFactory({ "xdp:xdp": xml });
+
+    expect(factory.numberPages).toEqual(1);
+
+    const pages = factory.getPages();
+    const a = searchHtmlNode(pages, "name", "a");
+    expect(a.attributes.href).toEqual("https://github.com/mozilla/pdf.js");
```

Comment:
Can you please add a check to ensure that the `target` is also correct, since you're using the "app.launchURL(...)" format to specify the URL?